### DISCUSSION TITLE
Rotate axis y-label

### DISCUSSION THREAD

**Discussion by cristobaltapia** [cetz-package/cetz-plot#66] [Q&A] [answered]
Hi,

is it possible to only rotate the y-label. I have found a way to rotate labels, but both x and y-labels are rotated. This is the code I have:

```typst
  #cetz.canvas({
    import cetz.draw: *
    set-style(axes: (
      label: (angle: 90deg),
    ))
    plot.plot(
      size: (10, 10),
      x-label: [x],
      y-label: [y],
      {
        plot.add(domain: (0, 200), x => 220 - 1.2 * x)
      },
    )

  })
```

**Comment by johannes-wolf** [comment, marked as answer]
Yes, you can set different style values for different axes/sides by using `left`, `bottom`, `right`, `top` in the `axes: (...)` style dict.

Example:
```typst
// --- >8 ---
set-style(axes: 
  (left: (label: (angle: 90deg))),
)
// --- 8< ---
```

**Comment by cristobaltapia** [comment]
Thanks! That works with the "scientific" axis-style, but when setting `axis-style: "left"`, it does not work. I suppose that this is a bug?

**Reply by johannes-wolf** [reply]
There you have to use `x` and `y` instead of `left` and `bottom` for styling.

**Reply by cristobaltapia** [reply]
I guess I answered myself at the same time as you did :P 

Thanks for the package! This might really allow me to move 100% to typst (...coming from many years of latex + tikz)

**Comment by cristobaltapia** [comment]
Oh, I got it! In that case you have to do

```typst
    set-style(
      axes: (
        y: (
          label: (angle: 90deg),
        ),
      ),
    )
```

So `y` instead of `left`. That is a bit confusing.
